app/src/settings/reset_settings_on_start.c

```diff
@@ -0,0 +1,19 @@
+/*
+ * Copyright (c) 2023 The ZMK Contributors
+ *
+ * SPDX-License-Identifier: MIT
+ */
+
+#include <zephyr/device.h>
+#include <zephyr/init.h>
+
+#include <zmk/settings.h>
+
+static int reset_settings_init(const struct device *dev) {
```

Comment:
This changed in Zephyr 3.5:

```suggestion
static int reset_settings_init(void) {
```

Probably should rebase to be sure this is 100% happy.